src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/grammar/java14/InputJava14LocalRecordAnnotation.java

```diff
@@ -0,0 +1,27 @@
+//non-compiled with javac: Compilable with Java14
+package com.puppycrawl.tools.checkstyle.grammar.java14;
+
+class InputJava14LocalRecordAnnotation {
+
+    public void doThing() {
+        // No error unless an annotation is present
+        @ProblemCauser
+        record Test(String hello, int world) {}
+
+        @ProblemCauser
+        class TestClass {}
+
+        @ProblemCauser @Deprecated
+        record Test2(String hello, int world) {}
+
+        @ProblemCauser @Deprecated
+        class TestClass2 {}
+
+        final var item = new Test("hi there", 22);
```

Comment:
Do we need this two lines?

Reply:
I usually try to keep things as close to the actual bug report as possible, but I can remove them if you want.

Reply:
I'm ok to keep this one.

Reply:
I just removed it anyway, I had to fix a spelling; I figured that I should get rid of it to prevent further CI runs. :)